src/alchemlyb/preprocessing/subsampling.py

```diff
@@ -97,14 +96,10 @@ def statistical_inefficiency(df, series=None, lower=None, upper=None, step=None)
 
         # calculate statistical inefficiency of series
         statinef  = statisticalInefficiency(series)
-
-        # we round up
-        statinef = int(np.rint(statinef))
-
-        # subsample according to statistical inefficiency
-        series = series.iloc[::statinef]
-
-        df = df.loc[series.index]
+        
+        #use the subsampleCorrelatedData function to get the subsample index
+        indices = subsampleCorrelatedData(series, g=statinef)
```

Comment:
Add explicit keywords for `subsampleCorrelatedData(..., fast=False, conservative=False, verbose=False)` (especially `conservative=False` to make sure that the behavior in alchemlyb won't change, even if pymbar changes its defaults.)

Reply:
Have we got tests that test fractional `g`?

Reply:
There is a test_preprocessing.py code, and inside there is a test_subsampling function https://github.com/alchemistry/alchemlyb/blob/0587ebd384b47eeb16e0d4c7acb1029c03b7ddda/src/alchemlyb/tests/test_preprocessing.py#L76 testing that the subsampled data is not bigger than the original data, while I think this one didn't explicitly test the `g` value or the actual size of the subsample data. Also, should we define a slicer function inside this class, maybe a bug here?